core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -123,11 +124,15 @@ Map<String, String> params() {
   }
 
   public RESTSessionCatalog() {
-    this(config -> HTTPClient.builder(config).uri(config.get(CatalogProperties.URI)).build());
+    this(config -> HTTPClient.builder(config).uri(config.get(CatalogProperties.URI)).build(), null);
   }
 
-  RESTSessionCatalog(Function<Map<String, String>, RESTClient> clientBuilder) {
+  public RESTSessionCatalog(
+      Function<Map<String, String>, RESTClient> clientBuilder,
+      BiFunction<Map<String, String>, Object, FileIO> ioBuilder) {
```

Comment:
I think that this should pass the `SessionContext` back rather than the wrapped identity. The session context was created by the caller anyway.

Reply:
Good idea, changed